stdlib/public/core/SetAlgebra.swift

```diff
@@ -137,7 +137,7 @@ public protocol SetAlgebra : Equatable, ExpressibleByArrayLiteral {
   /// - Note: if this set and `other` contain elements that are equal but
   ///   distinguishable (e.g. via `===`), which of these elements is present
   ///   in the result is unspecified.
-  func intersection(_ other: Self) -> Self
+  __consuming func intersection(_ other: __consuming Self) -> Self
```

Comment:
I think this argument either doesn't need to be consuming (and similarly for `formIntersection`), or `other` in `subtracting` below does, since intersection and subtraction seem like they behave in essentially the same way wrt to ownership (mutating the incoming collection). Thoughts?